accounts/views.py

```diff
@@ -0,0 +1,34 @@
+from django.contrib.auth import authenticate, get_user_model, login
+from django.http.response import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.urls import reverse, reverse_lazy
+from django.views.generic import CreateView
+
+from .forms import UserChangeForm, UserCreationForm
+from .models import User
+
+user = get_user_model()
+
+
+class SignUpView(CreateView):
+    form_class = UserCreationForm
+    model = User
+    template_name = 'accounts/signup.html'
+    success_url = reverse_lazy('twitter:homepage')
+
+    def form_valid(self, form):
+        form.save()
+        username = form.cleaned_data.get('username')
+        password = form.cleaned_data.get('password1')
+        user = authenticate(username=username, password=password)
+        if user is not None:
+                return HttpResponseRedirect(reverse_lazy('twitter:homepage'))
```

Comment:
form_validメソッドはデフォルトでsuccess_urlに飛ぶようになっているので、これをそのまま継承させてしまいましょう。また、前回書いていただいたlogin()メソッドは消さなくてOKです。ユーザー登録したらログインできていることを表すのはユーザー登録機能として含めてもらって構いません。前回述べたのは、ユーザー登録時以外でログイン・ログアウトしないようにして欲しかったということですね。
```suggestion
        if user is not None:
                login(self.request, user)
                return super().form_valid(form)
```
としてsuperメソッドを用いればOKです。